benchmarks/tosi_et_al_2015_gcubed/tosi.cc

```diff
@@ -0,0 +1,621 @@
+
+            if (this->convert_output_to_years() == true)
+              {
+                // fill statistics file
+                // make sure that the columns filled by this object
+                // all show up with sufficient accuracy and in scientific notation
+                statistics.add_value ("Viscous dissipation (J/yr)", viscous_dissipation * year_in_seconds);
+                statistics.set_precision ("Viscous dissipation (J/yr)", 8);
+                statistics.set_scientific ("Viscous dissipation (J/yr)", true);
+                statistics.add_value ("Rate of work (Km/yr)", work * year_in_seconds);
+                statistics.set_precision ("Rate of work (Km/yr)", 8);
+                statistics.set_scientific ("Rate of work (Km/yr)", true);
+                statistics.add_value ("Error", error);
+                statistics.set_precision ("Error", 8);
+                statistics.set_scientific ("Error", true);
+              }
+            else
+              {
+                // fill statistics file
+                // make sure that the columns filled by this object
+                // all show up with sufficient accuracy and in scientific notation
+                statistics.add_value ("Viscous dissipation (W)", viscous_dissipation);
+                statistics.set_precision ("Viscous dissipation (W)", 8);
+                statistics.set_scientific ("Viscous dissipation (W)", true);
+                statistics.add_value ("Rate of work (W)", work);
+                statistics.set_precision ("Rate of work (W)", 8);
+                statistics.set_scientific ("Rate of work (W)", true);
+                statistics.add_value ("Error", error);
+                statistics.set_precision ("Error", 8);
+                statistics.set_scientific ("Error", true);
+              }
+
+            std::ostringstream output;
+            output.precision(3);
+            if (this->convert_output_to_years() == true)
+              output << viscous_dissipation *year_in_seconds
+                     << " J/yr "
+                     << work *year_in_seconds
+                     << " J/yr "
+                     << error
+                     << " error";
+            else
+              output << viscous_dissipation
+                     << " W "
+                     << work
+                     << " W "
+                     << error
+                     << " error";
+
+            return std::pair<std::string, std::string> ("Viscous dissipation, rate of work, error:",
+                                                        output.str());
+    }
+
+
+  }
+}
+
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace TosiBenchmark
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(TosiMaterial,
+        "TosiMaterial",
+        "A material model that has constant values "
+        "for all coefficients but the density and viscosity as described in the "
```

Comment:
grammatically not wrong, but it's easy for people who are not used English speakers to understand it if you say "for all coefficients except the density..."